src/legacy/core_plugins/vega/public/plugin.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// @ts-ignore
+import { createVegaFn } from './vega_fn';
+// @ts-ignore
+import { createVegaTypeDefinition } from './vega_type';
+
+import { VegaSetupPlugin } from './setup';
+
+export class VegaPlugin {
```

Comment:
You can use type from core in your class https://github.com/streamich/kibana/blob/shim-embeddable/src/legacy/core_plugins/embeddable_api/public/shim/public/plugin.ts#L20